src/Features/Core/Portable/StackTraceExplorer/StackTraceExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    using StackFrameToken = EmbeddedSyntaxToken<StackFrameKind>;
+    using StackFrameTrivia = EmbeddedSyntaxTrivia<StackFrameKind>;
+
+    internal static class StackTraceExtensions
+    {
+        public static string CreateString(this StackFrameNode node, StackFrameTree tree)
+            => node.CreateString(tree, skipTrivia: false);
+
+        public static string CreateString(this StackFrameNode node, StackFrameTree tree, bool skipTrivia)
+        {
+            var span = skipTrivia
+                ? node.GetSpanWithoutTrivia(skipLeadingTrivia: true, skipTrailingTrivia: true)
+                : node.GetSpan();
+
+            return tree.Text.GetSubSequence(span).CreateString();
```

Comment:
so i feel like you shouldn't need the original tree or text.  nodes/tokens/trivia always contain enough to just build htings.